app/src/main/java/com/eveningoutpost/dexdrip/Models/blueReader.java

```diff
@@ -80,24 +110,37 @@ public static boolean isblueReaderPacket(byte[] buffer) {
             Log.i (TAG, "blueReader notice that NFC is now really hibernated...");
             if (counterHibernated > 0) {
                 Log.w (TAG,"Found hibernation after wrong read. Resend read-command...");
-                return new byte[]{0x6C};
+                return requestValue;
             } else {
                 return null;
             }
         } else if (bufferstring.startsWith("-r 0:")) {
-            Log.d (TAG, "blueReader sends an unknown reaktion: '" + bufferstring + "'");
+            Log.d (TAG, "blueReader sends an unknown reaction: '" + bufferstring + "'");
             return null;
         } else if (bufferstring.startsWith("TRANS_FAILED")) {
             Log.w (TAG, "Attention: check position of blueReader on the sensor, as it was not able to read!");
-            Home.toaststatic("check position of blueReader on the sensor, as it was not able to read!");
+                try {
+                    Home.toaststatic(xdrip.getAppContext().getString(R.string.bluereader_position));
+                } catch (Exception e) {
+                    Log.i (TAG, "TRANS_FAILED in test found");
+                }
+
             return null;
         } else if (bufferstring.startsWith("battery: ")) {
             if (BgReading.last() == null || BgReading.last().timestamp + (4 * 60 * 1000) < System.currentTimeMillis()) {
-                return new byte[]{0x6C};
+                return requestValue;
             }
         } else {
             counterHibernated = 0;
             processNewTransmitterData(TransmitterData.create(buffer, len, timestamp), timestamp);
+            // check for shutdown blueReader if Battery is too low
+            if (Pref.getBooleanDefaultFalse("blueReader_turn_off")) {
+                if (Pref.getInt("blueReader_turn_off_value",5) > Pref.getInt("bridge_battery",100)) {
+                    Log.w (TAG, "blueReader will be turn off, as the battery is lower then " + Pref.getInt("blueReader_turn_off_value",5) +"%");
+                    Home.toaststatic(xdrip.getAppContext().getString(R.string.bluereaderoff) + Pref.getInt("blueReader_turn_off_value",5) +"%");
+                    return shutdown;
+                }
+            }
```

Comment:
is this covered by tests?

Reply:
Not now, will implement it.

Reply:
@ahaarrestad the shutdown I have now included. But I have problems for the test if in the code is a (xdrip.getAppContext().getString(R.string.bluereaderoff) is used. is there a clean way to handle that in a test, or should I get that with a try-clause in the code?